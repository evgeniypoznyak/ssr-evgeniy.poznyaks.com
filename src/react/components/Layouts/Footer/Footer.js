import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import {Link} from 'react-router-dom';
import Avatar from '@material-ui/core/Avatar';
import {Paper} from '@material-ui/core';
import {Email, Phone} from '@material-ui/icons';

const useStyles = makeStyles({
    root: {
        marginTop: 20,
        // '@media (max-width: 960px)': {
        //     display: 'none',
        // },
    },
    menuWrapper: {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-evenly',
    },
    avatar: {
        height: 25,
        width: 25,
    },
});

export default function Footer() {
    const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);
    const scrollToTop = () => {
        window.scrollTo(0, 0);
    };
    const classes = useStyles();
    const [value, setValue] = React.useState(null);

    return (
        <Paper className={classes.root}>
            <BottomNavigation
                value={value}
                onChange={(event, newValue) => {
                    setValue(newValue);
                }}
                showLabels
                className={classes.menuWrapper}
            >
                <BottomNavigationAction
                    onClick={scrollToTop}
                    label="Email Me"
                    component={AdapterLink}
                    to={'/contact-us/'}
                    icon={<Email/>}
                />
                <BottomNavigationAction
                    label="Call Me"
                    href="tel:518-772-8217"
                    icon={<Phone/>}
                />
                <BottomNavigationAction
                    color="inherit"
                    target="_blank"
                    label="My GuiHub"
                    href={'https://github.com/evgeniypoznyak/'}
                    icon={<Avatar className={classes.avatar} alt={'My GitHub'} src={'/assets/github-grey.jpg'}/>}
                />
                <BottomNavigationAction
                    color="inherit"
                    target="_blank"
                    label="My LinkedIn"
                    href={'https://www.linkedin.com/in/evgeniypoznyak/'}
                    icon={<Avatar alt={'My LinkedIn'} className={classes.avatar} src={'/assets/linkedin-grey.jpg'}/>}
                />
            </BottomNavigation>
        </Paper>

    );
}
