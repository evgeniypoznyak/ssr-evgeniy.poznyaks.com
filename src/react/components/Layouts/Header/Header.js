import React, {Fragment, useContext} from 'react';
import {makeStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import PropTypes from 'prop-types';
import CssBaseline from '@material-ui/core/CssBaseline';
import useScrollTrigger from '@material-ui/core/useScrollTrigger';
import Button from '@material-ui/core/Button';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import InboxIcon from '@material-ui/icons/MoveToInbox';
import MailIcon from '@material-ui/icons/Mail';
import {NavLink} from 'react-router-dom';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import {Link} from 'react-router-dom';
import {State} from '../../../shared/StateManager';


function ElevationScroll(props) {
    const {children, window} = props;
    // Note that you normally won't need to set the window ref as useScrollTrigger
    // will default to window.
    // This is only being set here because the demo is in an iframe.
    const trigger = useScrollTrigger({
        disableHysteresis: true,
        threshold: 0,
        target: window ? window() : undefined,
    });

    return React.cloneElement(children, {
        elevation: trigger ? 4 : 0,
    });
}

ElevationScroll.propTypes = {
    children: PropTypes.node.isRequired,
    // Injected by the documentation to work in an iframe.
    // You won't need it on your project.
    window: PropTypes.func,
};

function Header(props) {
    const useStyles = makeStyles((theme) => ({
        root: {
            flexGrow: 1,
        },
        menuButton: {
            marginRight: theme.spacing(2),
        },
        title: {
            flexGrow: 1,
        },
        list: {
            width: 250,
        },
        fullList: {
            width: 'auto',
        },
        avatar: {
            margin: 10,
        },
        bigAvatar: {
            margin: 10,
            width: 60,
            height: 60,
        },
        skillName: {
            margin: 'auto',
        },
    }));

    const [state, setState] = React.useState({
        top: false,
        left: false,
        bottom: false,
        right: false,
    });
    const classes = useStyles();
    const context = useContext(State);
    const toggleDrawer = (side, open) => (event) => {
        if (event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
            return;
        }

        setState({...state, [side]: open});
    };
    const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);
    const leftSideList = (side) => (
        <div
            className={classes.list}
            role="presentation"
            onClick={toggleDrawer(side, false)}
            onKeyDown={toggleDrawer(side, false)}
        >
            <List>
                {/* todo replace this loop */}
                {['All mail', 'Trash', 'Spam'].map((text, index) => (
                    <ListItem button key={text}>
                        <ListItemIcon>{index % 2 === 0 ? <InboxIcon/> : <MailIcon/>}</ListItemIcon>
                        <ListItemText primary={text}/>
                    </ListItem>
                ))}
            </List>
            <Divider/>
            <List>
                <Typography variant={'h6'} align={'center'}>
                    My skills
                </Typography>
                {context.data.rawData.map((skill) => (
                    <Fragment key={skill.id}>
                        <ListItem
                            alignItems={'center'}
                            button
                            component={AdapterLink}
                            to={'/skills/' + skill.id}>
                            <ListItemAvatar>
                                <Avatar alt={skill.name} src={skill.logo}/>
                            </ListItemAvatar>
                            <ListItemText primary={skill.name} className={classes.skillName}/>
                        </ListItem>
                        <Divider component="li"/>
                    </Fragment>
                ))}
            </List>
        </div>
    );
    const rightSideList = (side) => (
        <div
            className={classes.list}
            role="presentation"
            onClick={toggleDrawer(side, false)}
            onKeyDown={toggleDrawer(side, false)}
        >
            <List>
                <Typography variant={'h6'} align={'center'}>
                    Admin menu
                </Typography>
                {/* todo replace this loop */}
                {['Right-Inbox', 'Right-Starred', 'Right-Send email', 'Right-Drafts'].map((text, index) => (
                    <ListItem button key={text}>
                        <ListItemIcon>{index % 2 === 0 ? <InboxIcon/> : <MailIcon/>}</ListItemIcon>
                        <ListItemText primary={text}/>
                    </ListItem>
                ))}
            </List>
            <Divider/>
            <List>
                {/* todo replace this loop */}
                {['All mail', 'Trash', 'Spam'].map((text, index) => (
                    <ListItem button key={text}>
                        <ListItemIcon>{index % 2 === 0 ? <InboxIcon/> : <MailIcon/>}</ListItemIcon>
                        <ListItemText primary={text}/>
                    </ListItem>
                ))}
            </List>
        </div>
    );

    const adminMenu = null;

    return (
        <React.Fragment>
            <CssBaseline/>
            <ElevationScroll {...props}>
                <AppBar>
                    <Toolbar>
                        <IconButton edge="start" className={classes.menuButton} onClick={toggleDrawer('left', true)}
                            color="inherit" aria-label="Menu">
                            <MenuIcon/>
                        </IconButton>
                        <Typography variant="h6" className={classes.title}>
                            <NavLink
                                to={'/'}
                                style={{textDecoration: 'none', color: 'inherit'}}
                                activeClassName={classes.active}>
                                <Button color="inherit">REACT SSR TEST</Button></NavLink>
                        </Typography>
                        <Button
                            href={'https://evgeniy.poznyaks.com/docs/resume/Evgeniy_Poznyak_Resume.pdf'}
                            color="inherit"
                            target="_blank"
                        >TEST</Button>
                        {adminMenu}
                    </Toolbar>
                </AppBar>
            </ElevationScroll>
            <Toolbar/>
            <Drawer open={state.left} onClose={toggleDrawer('left', false)}>
                {leftSideList('left')}
            </Drawer>
            <Drawer anchor="right" open={state.right} onClose={toggleDrawer('right', false)}>
                {rightSideList('right')}
            </Drawer>
        </React.Fragment>
    );
}

export default Header;
