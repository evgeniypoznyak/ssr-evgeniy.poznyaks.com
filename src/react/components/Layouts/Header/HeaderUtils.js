export function keyboardKeys(event) {
    return !(event && event.type && event.key)
        ? false
        : (event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift'));
}

export const processToggleDrawer = (setState, state) => {
    return (side, open) => {
        return event => {
            if (!keyboardKeys(event)) setState({...state, [side]: open});
        };
    };
};
