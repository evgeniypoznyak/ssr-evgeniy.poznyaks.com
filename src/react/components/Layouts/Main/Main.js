import React from 'react';
import {Grid} from '@material-ui/core';
import CenterPane from '../Panes/CenterPane/CenterPane';
import SkillPane from '../Panes/SkillPane/SkillPane';


const styles = {
    Paper: {
        marginTop: 5,
        marginBottom: 10,
    },
    CenterPane: {
        height: '100%',
        padding: 20,
        marginTop: 5,
        marginBottom: 10,
        marginLeft: 10,
        marginRight: 10,
    },
};

const Main = props => {
    return (
        <Grid container justify={'space-between'}>
            <SkillPane styles={styles} skills={props.context.data.sortedData.left}/>
            <CenterPane styles={styles}>{props.children}</CenterPane>
            <SkillPane styles={styles} skills={props.context.data.sortedData.right}/>
        </Grid>
    );
};

export default Main;
