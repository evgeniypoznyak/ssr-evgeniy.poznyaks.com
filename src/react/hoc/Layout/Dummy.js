import Button from '@material-ui/core/Button';
import React from 'react';

export const Dummy = props => {
    const dummyClick = () => {
        console.log('Clicked!');
    };

    return (<div >
        I'm dummy div
        <Button href={null} onClick={dummyClick}>I'm button</Button>
    </div>);
};