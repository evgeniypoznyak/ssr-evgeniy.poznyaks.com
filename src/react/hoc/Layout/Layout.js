import React, {Fragment} from 'react';
// import Navbar from '../../Navbar';
import Header from '../../components/Layouts/Header/Header';
import Main from '../../components/Layouts/Main/Main';
import Footer from '../../components/Layouts/Footer/Footer';

const Layout = (props) => {
    return (
        <Fragment>
            <Header/>
            <Main>
                {props.children}
            </Main>
            <Footer/>
        </Fragment>
    );
};
export default Layout;
