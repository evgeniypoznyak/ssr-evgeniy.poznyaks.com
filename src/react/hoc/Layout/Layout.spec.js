import React, {Fragment} from 'react';
import {shallow} from 'enzyme';
import Layout from './Layout';
import Header from '../../components/Layouts/Header/Header';
import Main from '../../components/Layouts/Main/Main';
import Footer from '../../components/Layouts/Footer/Footer';

let wrapper;

export function mockComponent(componentName) {
    return props => {
        return (
            <Fragment>{componentName}</Fragment>
        );
    };
}

jest.mock(
    '../../components/Layouts/Header/Header', () => {
        return mockComponent('Header');
    },
);
jest.mock(
    '../../components/Layouts/Main/Main', () => {
        return mockComponent('Main');
    },
);

beforeAll(() => {
    wrapper = shallow(<Layout>test</Layout>);
});

describe('<Layouts/>', () => {
    it('renders correctly', () => {
        expect(wrapper.children.length).toBe(1);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should render Header', () => {
        const header = wrapper.find(Header);
        expect(header).toHaveLength(1);
    });

    it('should render Main', () => {
        const header = wrapper.find(Main);
        expect(header).toHaveLength(1);
    });

    it('should render Footer', () => {
        const header = wrapper.find(Footer);
        expect(header).toHaveLength(1);
    });
});


