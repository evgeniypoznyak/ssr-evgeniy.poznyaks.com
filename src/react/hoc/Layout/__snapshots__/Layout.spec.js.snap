// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layouts/> renders correctly 1`] = `
<React.Fragment>
  <Unknown />
  <Unknown>
    test
  </Unknown>
  <Footer />
</React.Fragment>
`;
